File: Lab 4 Java/src/Project/Student.java
package Project;
import java.util.Scanner;

public class Student 
{
	private String name;
	private int no;
	   private double physics;
       private double programming;
       private double history;
	
	
	public String getName() { return name; }
	public int getNo() { return no; }
	
	 public double getPhysics() { return physics; }
     public double getProgramming() { return programming; }
     public double getHistory() { return history; }
     public double GetAverage() { return ((history + physics + programming) / 3); }
	public void setName(String nm) { this.name = nm; }
	
	public boolean setNo(int num)
	{
	if (num > 0)
	{
	no = num;
	return true;
	}
	else
	{
	no = 0;
	return false;
	}
	}
	
	public boolean setPhysics(double p)
    {
        if (p > 0 && p < 6)
        {
            physics = p;
            return true;
        }
        else
        {
            physics = 0;
            System.out.println("You have entered an incorrect estimate, please try again!");
            System.out.println("=========================================================");
            return false;
        }
    }
	
	public boolean setProgramming(double pr)
    {
        if (pr > 0 && pr < 6)
        {
            programming = pr;
            return true;
        }
        else
        {
            programming = 0;
            System.out.println("You have entered an incorrect estimate, please try again!");
            System.out.println("=========================================================");
            return false;
        }
    }

    public boolean setHistory(double h)
    {
        if (h > 0 && h < 6)
        {
            history = h;
            return true;
        }
        else
        {
            history = 0;
            System.out.println("You have entered an incorrect estimate, please try again!");
            System.out.println("=========================================================");
            return false;
           
        }
    }
    
	public boolean Init(String nm, int num, double p, double pr, double h)
	{
		  setName(nm);
          return setNo(num) && setPhysics(p) && setProgramming(pr) && setHistory(h);
    }
	
	public void Read()
	{
	Scanner scan = new Scanner(System.in); 
	
	String nm;
    int num;
    double p;
    double pr;
    double h;
	System.out.println("Name: ");
	nm = scan.nextLine();
	do
	{
	System.out.println("Number: ");
	num = scan.nextInt();
	System.out.println("Physics: ");
	p = scan.nextDouble();
	System.out.println("Programming: ");
	pr = scan.nextDouble();
	System.out.println("History: ");
	h = scan.nextDouble();
	} while (!Init(nm, num, p, pr, h));
	}
	
	public void Display()
	{
	System.out.println("===========================");
	System.out.println("Name: " + name);
	System.out.println("Number: " + no);
	System.out.println("Physics: " + physics);
	System.out.println("Programming: " + programming);
	System.out.println("History: " + history);
	System.out.println("===========================");
	}
	}

File: Lab 4/Lab 4/Student.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lab_4
{
    public class Student
    {
        private string name;
        private int no;
        private double physics;
        private double programming;
        private double history;

        public string GetName() { return name; }
        public int GetNo() { return no; }
        public double GetPhysics() { return physics; }
        public double GetProgramming() { return programming; }
        public double GetHistory() { return history; }

        public double GetAverage() { return ((history + physics + programming) / 3); }

        public void SetName(string nm) { this.name = nm; }
        public bool SetNo(int num)
        {
            if (num > 0)
            {
                no = num;
                return true;

            }
            else
            {
                no = 0;
                return false;
            }
        }

        public bool SetPhysics(double p)
        {
            if (p > 0 && p < 6)
            {
                physics = p;
                return true;
            }
            else
            {
                physics = 0;
                Console.WriteLine("You have entered an incorrect estimate, please try again!");
                Console.WriteLine("=========================================================");
                return false;
            }
        }

        public bool SetProgramming(double pr)
        {
            if (pr > 0 && pr < 6)
            {
                programming = pr;
                return true;
            }
            else
            {
                programming = 0;
                Console.WriteLine("You have entered an incorrect estimate, please try again!");
                Console.WriteLine("=========================================================");
                return false;
            }
        }

        public bool SetHistory(double h)
        {
            if (h > 0 && h < 6)
            {
                history = h;
                return true;
            }
            else
            {
                history = 0;
                Console.WriteLine("You have entered an incorrect estimate, please try again!");
                Console.WriteLine("=========================================================");
                return false;
               
            }
        }



        public bool Init(string nm, int num, double p, double pr, double h)
        {
            SetName(nm);
            return SetNo(num) && SetPhysics(p) && SetProgramming(pr) && SetHistory(h);
        }
        public void Read()
        {
            string nm;
            int num;
            double p;
            double pr;
            double h;
            Console.WriteLine("Name: ");
            nm = Convert.ToString(Console.ReadLine());
            do
            {
                Console.WriteLine("Number: ");
                num = Convert.ToInt32(Console.ReadLine()); // Int32 - звичайний int
                Console.WriteLine("Physics: ");
                p = Convert.ToDouble(Console.ReadLine());
                Console.WriteLine("Programming: ");
                pr = Convert.ToDouble(Console.ReadLine());
                Console.WriteLine("History: ");
                h = Convert.ToDouble(Console.ReadLine());


            } while (!Init(nm, num, p, pr, h));
        }



        public void Display()
        {
            Console.WriteLine("======================");
            Console.WriteLine($"Name: { name}");
            Console.WriteLine($"Number: {no}");
            Console.WriteLine($"Physics: { physics}");
            Console.WriteLine($"Programming: {programming}");
            Console.WriteLine($"History: {history}");
            Console.WriteLine("======================");


        }
    }
}